modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/ClaimDataMigrator.java

```diff
@@ -0,0 +1,293 @@
+ */
+public class ClaimDataMigrator extends Migrator {
+
+    private static final Log log = LogFactory.getLog(ClaimDataMigrator.class);
+
+    private static final String CLAIM_CONFIG = "claim-config.xml";
+
+    private ClaimConfig claimConfig;
+
+    private ClaimDialectDAO claimDialectDAO = new ClaimDialectDAO();
+
+    private CacheBackedLocalClaimDAO localClaimDAO = new CacheBackedLocalClaimDAO(new LocalClaimDAO());
+
+    private CacheBackedExternalClaimDAO externalClaimDAO = new CacheBackedExternalClaimDAO(new ExternalClaimDAO());
+
+    @Override
+    public void migrate() throws MigrationClientException {
+
+        String filePath = Utility.getDataFilePath(CLAIM_CONFIG, getVersionConfig().getVersion());
+        try {
+            claimConfig = FileBasedClaimBuilder.buildClaimMappingsFromConfigFile(filePath);
+        } catch (IOException | XMLStreamException | UserStoreException e) {
+            String message = "Error while building claims from config file";
+            if (isContinueOnError()) {
+                log.error(message, e);
+            } else {
+                throw new MigrationClientException(message, e);
+            }
+        }
+
+        if (claimConfig.getClaims().isEmpty()) {
+            log.info(Constant.MIGRATION_LOG + "No data to migrate related with claim mappings.");
+            return;
+        }
+
+        try {
+            // Migrate super tenant.
+            migrateClaimData(Constant.SUPER_TENANT_ID);
+
+            // Migrate other tenants.
+            List<Tenant> tenants = Utility.getTenants();
+            List<Integer> inactiveTenants = Utility.getInactiveTenants();
+            boolean ignoreForInactiveTenants = isIgnoreForInactiveTenants();
+            for (Tenant tenant : tenants) {
+                int tenantId = tenant.getId();
+                if (ignoreForInactiveTenants && inactiveTenants.contains(tenantId)) {
+                    log.info("Skipping claim data migration for Inactive tenant : " + tenantId);
+                    continue;
+                }
+                migrateClaimData(tenant.getId());
+            }
+        } catch (UserStoreException | ClaimMetadataException e) {
+            String message = "Error while migrating claim data";
+            if (isContinueOnError()) {
+                log.error(message, e);
+            } else {
+                throw new MigrationClientException(message, e);
+            }
+        }
+    }
+
+    /**
+     * Migrate claim mappings.
+     *
+     * @param tenantId tenant id
+     * @throws UserStoreException     UserStoreException
+     * @throws ClaimMetadataException ClaimMetadataException
+     */
+    private void migrateClaimData(int tenantId) throws UserStoreException, ClaimMetadataException {
+
+        UserRealm realm = ISMigrationServiceDataHolder.getRealmService().getTenantUserRealm(tenantId);
+        String primaryDomainName = realm.getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig
```

Comment:
Is there a possibility of realm attribute having a null value for an invalid tenant id? If so handle null case to avoid NPE.